### PR TITLE
fix CI failures for e2e tests

### PR DESCRIPTION
**Pull request by rabi-siddique**
Recently, the e2e tests across all dApps have been consistently failing. The failure happens where Cypress appears to use IPv6, causing the tests to fail in the CI setup. The underlying infrastructure of GitHub Actions may have different network configurations across different runs. Sometimes, the environment might resolve `localhost` to `127.0.0.1 (IPv4)`, while other times it might resolve to `::1 (IPv6)`, leading to inconsistencies in the connection check.

The solution implemented in this PR sets the environment variable `NODE_OPTIONS=--dns-result-order=ipv4first` to prioritize IPv4 DNS resolution.

### CODE REVIEW

File: .github/workflows/reusable-workflow.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ jobs:
 
       - name: Upload e2e artifacts as workflow artifacts
         uses: actions/upload-artifact@v4
-        if: ${{ inputs.is_emerynet_test && !cancelled() }}
+        if: ${{ !inputs.is_emerynet_test && !cancelled() }}
```
Comment (frazarshad) [inline .github/workflows/reusable-workflow.yml:105]: i think this should be reverted

File: test/e2e/docker-compose-reconstitution.yml [hunk 2]
```diff
@@ -9,12 +9,27 @@ services:
...
+        echo -n "======> remote noVNC URL: " &&
+        curl -s ngrok:4040/api/tunnels | jq -r .tunnels[0].public_url &&
+        
+        # Start nginx if running locally
+        if [ "$CYPRESS_AGORIC_NET" == "local" ]; then
+          nginx
+        fi &&
+        
+        # Start development server and wait for it to be ready
+        yarn dev & sleep 10 &&
```
[the remark this answers is not in the document]
Comment (rabi-siddique) [reply test/e2e/docker-compose-reconstitution.yml]: Already tried and tested. Please see this:
https://github.com/cypress-io/github-action?tab=readme-ov-file#wait-on-with-nodejs-18

File: .github/workflows/reusable-workflow.yml (hunk 1 above)
Comment (rabi-siddique) [reply .github/workflows/reusable-workflow.yml:105]: No, it should be retained. If tests are not for emerynet, we should retain artifacts.

File: Dockerfile [hunk 3]
```diff
@@ -2,6 +2,8 @@ FROM ghcr.io/agoric/agoric-sdk:latest
 
 # Add the Agoric CLI to the PATH to access 'agops' from the shell.
 ENV PATH="/usr/src/agoric-sdk/packages/agoric-cli/bin:${PATH}"
+# Prefer IPv4 for DNS resolution
+ENV NODE_OPTIONS=--dns-result-order=ipv4first
```
[the remark this answers is not in the document]
Comment (rabi-siddique) [reply Dockerfile:6]: Refer to the link I shared above.

File: .github/workflows/reusable-workflow.yml (hunk 1 above)
Comment (frazarshad) [reply .github/workflows/reusable-workflow.yml:105]: right. my bad

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (frazarshad) [reply Dockerfile:6]: the link suggests doing this in conjunction with using wait-on. if we're not using wait-on and opting to use sleep instead, do we need this?

[the remark this answers is not in the document]
Comment (rabi-siddique) [reply Dockerfile:6]: Yes we need this. I've updated the PR description according to this solution. Please check.

**Review by frazarshad — APPROVED** [review]

**Pull request merged**